chapters/strings/count.md

```diff
@@ -0,0 +1,21 @@
+---
+layout: recipe
+title: Count Substring
+chapter: Strings
+---
+## Problem
+
+You need to count occurrence of a search string within a message.
+
+## Solution
+
+{% highlight coffeescript %}
+count = (substr, string) ->
+  num = pos = 0
+  return 1/0 unless substr.length
```

Comment:
Why perform a calculation when you could return `null` or `NaN`? Or `throw new TypeError("Must be a string") unless substr == String(substr)`.

Actually thinking about it `substr = String(substr)` would work perfect as it would always be a string (and if you pass in garbage the return is 0 (garbage)).

Also how do you feel about `searchTerm` or just `term` instead of `substr` which conflicts with the method `.substr()` which some readers might find confusing.